agent-inject/agent/annotations.go

```diff
@@ -519,6 +526,14 @@ func Init(pod *corev1.Pod, cfg AgentConfig) error {
 		pod.ObjectMeta.Annotations[AnnotationTemplateConfigStaticSecretRenderInterval] = cfg.StaticSecretRenderInterval
 	}
 
+	if _, ok := pod.ObjectMeta.Annotations[AnnotationTemplateConfigMaxConnectionsPerHost]; !ok {
+		if cfg.MaxConnectionsPerHost == 0 {
+			cfg.MaxConnectionsPerHost = DefaultTemplateConfigMaxConnectionsPerHost
+		}
+
+		pod.ObjectMeta.Annotations[AnnotationTemplateConfigMaxConnectionsPerHost] = strconv.FormatInt(cfg.MaxConnectionsPerHost, 10)
```

Comment:
I'm not that familiar with the parseutil package, so I'm curious why it's being used some places and not others (like this line).

Reply:
I believe that we're trying to make an effort on using `parseutil` within `go-secure-stdlib` for arbitrarily-typed values. For example, [ParseInt](https://github.com/hashicorp/go-secure-stdlib/blob/b4a7f3e5652cb93b2c3c3f42bb2b2c8ed21641ac/parseutil/parseutil.go#L215) handles strings, `json.Number`, and various integer types. I do not believe that `go-secure-stdlib` has an analogous function to format integers as strings. In this case, we're providing a hardcoded scalar value so it is more deterministic.

Reply:
It looks like `parseutil.ParseString()` may work here? But good point about it not being arbitrary input at this point, that makes sense.

Reply:
Oh that's a good point. It makes sense to remain consistent since that was the whole point of the change.

Reply:
Actually, on second thought, modifying it requires additional weak decode and error handling so I will leave it as is.